builtin/checkout-index.c

```diff
@@ -247,6 +253,8 @@ int cmd_checkout_index(int argc, const char **argv, const char *prefix)
 
 		if (all)
 			die("git checkout-index: don't mix '--all' and explicit filenames");
+		if (include_sparse)
+			die("git checkout-index: don't mix '--sparse' and explicit filenames");
```

Comment:
Why shouldn't these be combined? What if I want to poke at a file outside of my sparse-checkout? It certainly seems like `git checkout-index --sparse -- deep/deeper2/a` would be a good way to get that file on-disk.

Reply:
The reason I didn't include `--sparse` for individual files is because, as it is right now, individually-specified files will always expand the index if that specified file is outside the checkout definition. I wasn't really sure what the "right" way to handle it was, though. On one hand, not requiring a `--sparse` flag to `checkout-index` individual files (even those outside the checkout definition) is the most consistent with existing behavior. On the other hand, it is kind of weird that we need a special `--sparse` option for files in sparse directories for `--all`, but not when you list out individual files.

What do you think would be better (especially based on how you've handled similar situations in other commands)?

Reply:
Since this is a plumbing command, let's be cautious and avoid the behavior break by requiring `--sparse` for individual files. (This is a change of opinion from my earlier comment.)